smg-automotive/components-pkg
# refactor(tokens): Export plain themes and render tokens based on current theme

References: 
- https://github.com/smg-automotive/components-pkg/pull/59
- https://autoricardo.atlassian.net/browse/FE-31
- https://automotivesmg.slack.com/archives/C03GK939L3V/p1654246423987809

## Motivation and context

After reviewing the PR for our first component, btw great work y'all. I have been digging a bit into the setup we have so far. Besides some concrete points on our first component, after checking back with Virginia, I would like to make two suggestions that may make our lives easier in the future if we tackle them in the early stage of our new and 🌞 components library. 

1. Desing tokens showcase

I suggest showing the tokens based on the selected theme instead of hardcoding them. This will make the tokens represent the actual theme. Further on we don't need to adapt the stories when extending or updating the tokens. (For example, we missed adding the primary colour token when we added it, as well as for the new button we did not show the shadow token that is about to be added)

As a showcase, I refactored the colours part (and exposed some themes, see 2) to show what it would look like.
The themes we currently expose are "Moto Scout 24 with Chakra" and "Auto Scout 24 with Chakra"
![image](https://user-images.githubusercontent.com/3371760/171882328-9cba4f26-d153-4a35-a697-d98ee18b761c.png)

https://user-images.githubusercontent.com/3371760/171883795-97305a93-063b-4ec7-950c-f0c627824574.mov

As you can see in the video we actually export more colours (applies the same for other tokens) as we merge our custom theme with the default chakra one. 

2. Themes / Exports

We started by merging our Themes with the chakra one. I would remove and expose only the customer Them prior to using the components package in the consumer project. We may consider keeping the chakra merged version for showcasing purposes of the chakra components, I would avoid exporting them.

3. shared vs foundations

Chakra is using the term "foundations" for what we added in the "shared" folder, for consistency we might wanna consider renaming the folder 
ref: https://github.com/chakra-ui/chakra-ui/tree/main/packages/theme

---

Let me know your thoughts, cheers!

## How to test

Check the tokens in the storybook, use the provided themes to check the differences

### Update:

After the basic agreement, I went ahead and refactored the tokens. I added a simple table definition to we can use the chakra table in the storybook to showcase the tokens. We might wanna adapt the table definition before starting to use it in a project.
